torch/distributed/_tensor/op_schema.py

```diff
@@ -82,12 +82,17 @@ class PlacementStrategy:
     redistribute_cost: Optional[List[List[float]]] = None
 
     @cached_property
-    def out_spec(self) -> DTensorSpec:
-        if isinstance(self.output_spec, DTensorSpec):
-            return self.output_spec
+    def output_spec(self) -> DTensorSpec:
+        """
+        This function requires that the strategy have at least one DTensorSpec as the
+        output spec. If the output_specs is a tuple, then we will just return the first
+        element in the tuple and this element must be a DTensorSpec.
+        """
+        if isinstance(self.output_specs, DTensorSpec):
+            return self.output_specs
         else:
-            assert len(self.output_spec) > 0, "empty output_spec!"
-            spec = self.output_spec[0]
+            assert len(self.output_specs) > 0, "empty output_specs!"
+            spec = self.output_specs[0]
```

Comment:
I feel returning the first spec if it's a tuple it's kind of evil as it silently ignore other output specs. 

What was the motivation of handling this? I feel we can simply throw an ValueError here saying sth like trying to access `output_spec` of a placement strategy but found multiple output specs.

Reply:
So far the only operator that can produce a `output_specs` of type `Tuple[DTensorSpec, ...]` is backward layer norm. If we disable the `output_spec` function on the `Tuple[DTensorSpec, ...]` case by throwing error, IIUC it will throw error when layer norm back propagates the derivative to the next operator in backward pass. So I chose to return the output spec of `d_input` (the first in `output_specs`).

Reply:
do you mean for layernorm_backward where there would be outputs like (DTensorSpec, None, None)? 

For this case shouldn't we never try to access `placement_strategy.output_spec` and developer always need to access with `placement_strategy.output_specs`?

My understanding is that `placement_strategy.output_spec` is just a convenient util/property func that developer calls, and for the case where it returns multiple output_specs, they should not use `output_spec`

Reply:
sync offline.